lms/djangoapps/shoppingcart/views.py

```diff
@@ -691,6 +691,16 @@ def postpay_callback(request):
         # See if this payment occurred as part of the verification flow process
         # If so, send the user back into the flow so they have the option
         # to continue with verification.
+
+        # Only orders where order_items.count() == 1 might be attempting to upgrade
+        attempting_upgrade = request.session.get('attempting_upgrade', False)
+        if attempting_upgrade:
+            course_id = result['order'].orderitem_set.all().select_subclasses()[0].course_id
```

Comment:
I think this code should be more defensive.  What if there isn't an order item in the order, or the first order item doesn't have a course ID?